src/rules/no-redundant-files.ts

```diff
@@ -0,0 +1,272 @@
+import { createRule } from "../createRule.js";
+import { isJSONStringLiteral, isNotNullish } from "../utils/predicates.js";
+
+const defaultFiles = [
+	/* cspell:disable-next-line */
+	"LICENCE",
+	/* cspell:disable-next-line */
+	"LICENCE.md",
+	"LICENSE",
+	"LICENSE.md",
+	"package.json",
+	"README.md",
+] as const;
+
+const cachedRegex = new Map<string, RegExp>();
+const getCachedLocalFileRegex = (filename: string) => {
+	// Strip the leading `./`, if there is one, since we'll be incorporating
+	// it into the regex.
+	const baseFilename = filename.replace("./", "");
+	let regex = cachedRegex.get(baseFilename);
+	if (regex) {
+		return regex;
+	} else {
+		regex = new RegExp(`^(./)?${baseFilename}$`, "i");
+		cachedRegex.set(baseFilename, regex);
+		return regex;
+	}
+};
+
+/**
+ * A function to use as a fix, that will remove an element from an array of elements
+ * and manage the removal of any commas that would be necessary to maintain proper
+ * json.  If this is the last entry, and we don't have a comma (strict json), then
+ * we don't need to worry about removing this element's comma, but we should remove
+ * the comma from the element before, since it will the last element after the removal.
+ * If this is not the last line, and it's not the only line (which wouldn't have a comma)
+ * then we need to remove the comma from this line, so that it's not left behind after
+ * the node removal.
+ * @param elements All file entry elements
+ * @param currentIndex The index of the entry that we want to remove
+ * @param fixer The eslint fixer function we'll use to remove the entry
+ * @param context Current context
+ * @returns The list of fixes that the `context.report` fix needs to perform the removals for us
+ */
+const removeEntryAndAppropriateCommas = (
+	elements: JsonAST.JSONStringLiteral[],
+	currentIndex: number,
+	fixer: Rule.RuleFixer,
+	context: Rule.RuleContext,
+): Rule.Fix[] => {
+	const element = elements[currentIndex];
+	const fixes = [fixer.remove(element as unknown as ESTree.Node)];
+
+	// If this is not the last entry, then we need to remove the comma from this line.
+	// Or if it's jsonc, then we may need to remove the comma, even if it is the last line
+	const commaTokenFromCurrentLine =
+		context.sourceCode.getTokenAfter(element as unknown as ESTree.Node)
+			?.value === ","
+			? context.sourceCode.getTokenAfter(
+					element as unknown as ESTree.Node,
+				)
+			: null;
+	if (commaTokenFromCurrentLine) {
+		fixes.push(fixer.remove(commaTokenFromCurrentLine));
+	}
+
+	// If this is the last line and it's not the only entry, then the line above this one
+	// will become the last line, and should not have a trailing comma.
+	if (currentIndex > 0) {
+		const commaTokenFromPreviousLine =
+			context.sourceCode.getTokenAfter(
+				elements[currentIndex - 1] as unknown as ESTree.Node,
+			)?.value === ","
+				? context.sourceCode.getTokenAfter(
+						elements[currentIndex - 1] as unknown as ESTree.Node,
+					)
+				: null;
```

Comment:
[Testing] Looks like this isn't covered by unit tests. Could you please add in test coverage anywhere that needs it?

Reply:
This line went away with the above refactor.